html/user/server_status.php

```diff
@@ -187,9 +187,9 @@ function show_status_html($x) {
     end_table();
     
     global $server_version;
-    global $git_commit;
-    if ( isset($server_version) && isset($git_commit) ) {
-        echo "Server version: $server_version (<a href=https://github.com/BOINC/boinc/commit/$git_commit>$git_commit</a>)<br>";
+    global $git_url;
+    if ( isset($server_version) && isset($git_url) ) {
```

Comment:
I recommend to make the link optional but always show the version (if available). Right now `$git_url` is set in a way that makes the following assumptions:
- the source repo was cloned via `https`, not `ssh`
- the source repo's relevant remote is called "origin"

I think that both assumptions shouldn't be made. At least the code should prepare for the case that they're no met, i.e. make the link/URL optional and don't just assume you got a valid `$git_url`.

Reply:
The release.inc file can always be edited/deleted.

Reply:
What's the point of this PR if projects are meant to edit `release.inc` manually afterwards? What I'm trying to convey here is that the current implementation makes assumptions that won't always be met. Even worse, it doesn't handle gracefully the case when the assumptions aren't met, producing an invalid HTML link.